### PR TITLE
Option number 2

### LINKED ISSUE #29: Option No. 2 - Change algorithm settings

### PR DESCRIPTION
**Pull request by OmerShoulstein**
Fixes #29

### CODE REVIEW

File: ex3/server/changeAlgoSettings.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        ChebyshevDistance<T> chebyshevDistance;
+        ManhattanDistance<T> manhattanDistance;
+        EuclideanDistance<T> euclideanDistance;
+        std::string inputDistance = parameters[1];
+        if (inputDistance == "CHE" || inputDistance == "che") {
+            newDistance = &chebyshevDistance;
+        } else if (inputDistance == "MAN" || inputDistance == "man") {
+            newDistance = &manhattanDistance;
+        } else if (inputDistance == "EUC" || inputDistance == "euc") {
+            newDistance = &euclideanDistance;
```
Comment (TomBenDor) [inline ex3/server/changeAlgoSettings.h:52]: Turn inputDistance into lowercase and then compare:
https://stackoverflow.com/questions/313970/how-to-convert-an-instance-of-stdstring-to-lower-case

File: ex3/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,8 @@ set(CMAKE_CXX_STANDARD 11)
 
 add_executable(Server server/main.cpp server/KnnClassifier.h server/Classifiable.h server/Iris.h
         server/Iris.cpp utils.h utils.cpp server/TCPServer.cpp server/TCPServer.h server/Distance.h
-        server/EuclideanDistance.h server/ChebyshevDistance.h server/ManhattanDistance.h server/Command.h server/DefaultIO.h server/Command.cpp server/StandardIO.cpp server/StandardIO.h)
+        server/EuclideanDistance.h server/ChebyshevDistance.h server/ManhattanDistance.h server/Command.h
+        server/DefaultIO.h server/Command.cpp server/StandardIO.cpp server/StandardIO.h server/changeAlgoSettings.h
```
Comment (TomBenDor) [inline ex3/CMakeLists.txt:10]: Should ne ChangeAlgoSettings

File: ex3/server/Command.h [hunk 3]
```diff
@@ -9,7 +9,9 @@ class Command {
     std::string description;
     DefaultIO *io;
 public:
-    virtual std::string getDescription() = 0;
+    virtual std::string getDescription();
+
+    virtual DefaultIO *getDefaultIO();
```
Comment (TomBenDor) [inline ex3/server/Command.h:14]: Why it's not abstract?
```suggestion
    virtual std::string getDescription() = 0;

    virtual DefaultIO *getDefaultIO() = 0;
```

**Review by TomBenDor — CHANGES_REQUESTED** [review]

File: ex3/server/Command.h (hunk 3 above)
Comment (OmerShoulstein) [reply ex3/server/Command.h:14]: because it is implemented

**Review by TomBenDor — APPROVED** [review]

**Pull request merged**